lib/ueberauth/strategy/google.ex

```diff
@@ -130,15 +125,21 @@ defmodule Ueberauth.Strategy.Google do
     end
   end
 
-  defp with_param(opts, key, conn) do
-    if value = conn.params[to_string(key)], do: Keyword.put(opts, key, value), else: opts
-  end
-
-  defp with_optional(opts, key, conn) do
-    if option(conn, key), do: Keyword.put(opts, key, option(conn, key)), else: opts
+  @doc false
+  defp redirect_options(conn) do
```

Comment:
Regardless of whether or not it adds a significant amount of time to the process it's an unnecessary conversion to string and map lookup.  I'd like to find a better solution.